examples/minidump_stackwalk/src/main.rs

```diff
@@ -108,26 +114,50 @@ impl<'a> LocalSymbolProvider<'a> {
                     Err(_e) => continue,
                 };
 
-                let id = object.debug_id();
-
-                let object_list = object_db.entry(id).or_insert_with(Vec::new);
                 tracing::trace!(
                     object.path = ?entry.path(),
+                    object.code_id = ?object.code_id(),
+                    object.debug_id = ?object.debug_id(),
                     object.has_unwind_info = object.has_unwind_info(),
                     object.has_symbol_info = object.has_debug_info(),
                     "object found"
                 );
-                object_list.push(ObjectMetadata {
+
+                let object_meta = ObjectMetadata {
                     path: entry.path().into(),
                     index_in_archive: idx,
                     has_unwind_info: object.has_unwind_info(),
                     has_symbol_info: object.has_debug_info(),
-                })
+                };
+
+                let debug_id = object.debug_id();
+                if !debug_id.is_nil() {
+                    let by_debug_id = object_db
+                        .by_debug_id
+                        .entry(debug_id)
+                        .or_insert_with(Vec::new);
+                    by_debug_id.push(object_meta.clone());
+                }
+
+                if let Some(code_id) = object.code_id() {
+                    let by_code_id = object_db.by_code_id.entry(code_id).or_insert_with(Vec::new);
+                    by_code_id.push(object_meta);
```

Comment:
I wonder why we need to clone in `debug_id` case, but not here?

Reply:
Because we don’t need the original anymore after this call, so it is moved ;-)